### PR TITLE
Add go-links extension

### PR DESCRIPTION
**Pull request by bradgarropy**
## Description

Many companies have `go` links, a short human readable URL that redirects to another page. For example, typing in the following URL might redirect to the human resources website.

```
go/benefits
```

This integrates `go` links with Raycast so you can save them, give them descriptions, and search them right from the toolbar.

## Screencast

https://github.com/user-attachments/assets/0466ef0b-374a-4517-9f12-07fd825db110

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
We had this in the [queue](https://github.com/raycast/extensions/pull/10229) for a very long time. Maybe there is something you could use to improve it further?

File: extensions/go-links/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Changelog
```
Comment (pernielsentikaer) [inline extensions/go-links/CHANGELOG.md:1]: Changelog should look something [like this](https://raw.githubusercontent.com/raycast/extensions/9cf6a66f16f53e069a5032193b9b930bf52f1237/extensions/tableplus/CHANGELOG.md)

**Comment by pernielsentikaer** [comment]
It might not be the same, could you add some more in README about how to use it and if any tools is required

**Comment by bradgarropy** [comment]
> We had this in the [queue](https://github.com/raycast/extensions/pull/10229) for a very long time. Maybe there is something you could use to improve it further?

@pernielsentikaer Ah, so the extension you mentioned above uses a service called [golinks.io](https://www.golinks.io/) to manage all of the links. It allows Raycast users to enter their GoLinks API key and directly create links there.

Stripe has an internal implementation of Go links, and my extension just saves go links that you're interested on your local machine. There may be thousands of go links at a company, but it's like your personal collection of "starred" go links that you can search and open.

File: extensions/go-links/CHANGELOG.md (hunk 1 above)
Comment (bradgarropy) [reply extensions/go-links/CHANGELOG.md:1]: The [Version History](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history) section of your documentation references [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), which is the format that I followed here. Is that not accurate anymore?

Screenshot 2024-10-01 at 8 38 44 AM

**Comment by bradgarropy** [comment]
Someone just showed me an internal tool that does something very similar. So I'm okay with closing this one out!

**Pull request closed without merging**